### PR TITLE
backend/sqlstore: move GetDashboards to Dashboard Service

### PR DESCRIPTION
**Pull request by mildwonkey**
I also did some mild file renaming to try and get the dashboards package closer in line with the sqlstore split design doc.

Note that this PR is currently targeting another branch and will be updated to target main when #49170 is merged.

### CODE REVIEW

**Comment by mildwonkey** [comment]
This will updated to target main once 49170 and I'm pretty sure the drone tests will pass then, so this is ready for review but won't merge until it's targeting main.

**Comment by ying-jeanne** [comment]
Code seems good, probably need to check enterprise whether it is broken change or not. will approve when you change the target.

**Review by ying-jeanne — APPROVED** [review]

**Pull request merged**
